tests/rbd_mirror/rbd_mirror_utils.py

```diff
@@ -304,10 +311,10 @@ def initial_mirror_config(self, mirror2, poolname, imagename, **kw):
             p.spawn(mirror2.create_pool, poolname=poolname)
 
         self.create_image(imagespec=imagespec, size=imagesize)
-        if kw.get("mirrormode") != "snapshot":
-            self.image_feature_enable(imagespec=imagespec, image_feature="journaling")
```

Comment:
Can you please check now ?

Reply:
1) the existing condition `if kw.get("mirrormode") != "snapshot":` disallows user to use enable feature if in case of snapshot based mirroring
with my changes, user can now enable journaling if hournal mirror mode, or pass some value to image_features and he can enable features in snapshot based mirroring also right ?
(not that I see any usecases, but just adding inclusive)
2) imo, the defaulting could be done in rbd_mirror_config where we call current initial config method

Reply:
I am not very pro with defaulting mirrormode to journal in rbd_mirror_config, because the RBD cli by default doesn't expect mirrormode to be set for journal based mirroring. We would be failing to test this scenario in RBD cli if we default the value to journal in rbd_mirror_config. So in view of covering one condition, we would be leaving out one test scenario.

So my option would be to do something like this.

The below condition would enable journaling feature by default for journal based mirroring.
```
if kw.get("mirrormode") != "snapshot" and "journaling" not in kw.get("image_feature"): 
    self.image_feature_enable(imagespec=imagespec, image_feature="journaling")
```

The below condition would enable all features specified in kw, irrespective of the mirrormode.
```
if kw.get("image_feature"):
    self.image_feature_enable(imagespec=imagespec, image_feature=image_feature)
```

IMO this above code should cover all our scenarios.

Reply:
By word defaulting my thoughts were confined to cephci, got your perspective of matching  defaults with rbd cli after the offline syncup with you.

Thanks for the code also.
Have made changes accordingly.

Reply:
LGTM